src/test/groovy/com/twcable/grabbit/jcr/DefaultProtoNodeDecoratorSpec.groovy

```diff
@@ -0,0 +1,145 @@
+package com.twcable.grabbit.jcr
+
+import com.day.cq.commons.jcr.JcrConstants
+import com.twcable.grabbit.proto.NodeProtos
+import com.twcable.grabbit.proto.NodeProtos.Node as ProtoNode
+import com.twcable.grabbit.proto.NodeProtos.Node.Builder as ProtoNodeBuilder
+import com.twcable.grabbit.proto.NodeProtos.Property as ProtoProperty
+import spock.lang.Specification
+
+import javax.jcr.Node
+import javax.jcr.Property
+import javax.jcr.PropertyIterator
+import javax.jcr.Session
+
+import static javax.jcr.PropertyType.STRING
+import static org.apache.jackrabbit.JcrConstants.JCR_MIXINTYPES
+import static org.apache.jackrabbit.JcrConstants.JCR_PRIMARYTYPE
+
+/*
+ * Copyright 2015 Time Warner Cable, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class DefaultProtoNodeDecoratorSpec extends Specification {
+
+    ProtoNode decoratedProtoNode
+    ProtoProperty mixinProperty
+    ProtoProperty someOtherProperty
+
+    def setup() {
```

Comment:
I think as long as we are just reading these values and not doing any manipulation, we should be using setupSpec..